app/services/ab_test_bucket.rb

```diff
@@ -0,0 +1,61 @@
+class AbTestBucket
+  include ActiveModel::Model
+
+  attr_reader :buckets
+
+  validate :within_100_percent
+
+  def initialize(buckets: { default: 100 })
+    @buckets = buckets
+    raise 'invalid bucket data structure' unless valid_bucket_data_structure?
+    ensure_numeric_percentages
+  end
+
+  def bucket(discriminator = nil)
+    return :misconfigured unless valid?
```

Comment:
I would consider moving the validation elsewhere? like just have the constructor raise if things don't add up?

Reply:
Agreed!